esp-hal/src/dma/sdio.rs

```diff
@@ -0,0 +1,468 @@
+//! SDIO DMA types for SoCs with a dedicated engine.
```

Comment:
I took a look at the TRM. The SDIO's DMA is different enough to warrant it going into `src/sdio/dma.rs` instead of `src/dma/sdio.rs`.
I don't want any users getting confused when the buffer types and traits in the dma module don't work with the SDIO.

Reply:
Sounds good, I can move this code into the main SDIO PR.